cadence/contracts/AxelarGateway.cdc

```diff
@@ -0,0 +1,312 @@
+import AxelarAuthWeighted from "./auth/AxelarAuthWeighted.cdc"
+import Crypto
+
+// Main GateWay contract for audit
+access(all) contract AxelarGateway {
+  access(self) let SELECTOR_APPROVE_CONTRACT_CALL: [UInt8]
+  access(self) let SELECTOR_TRANSFER_OPERATORSHIP: [UInt8]
+  access(all) let PREFIX_APP_CAPABILITY_NAME: String
+
+  access(all) resource CGPCommand {
+    access(all) var commandId: String
+    access(all) var isExecuted: Bool
+
+    destroy() {
+      emit Executed(commandId: self.commandId)
+    }
+
+    init(commandId: String) {
+      self.commandId = commandId
+      self.isExecuted = false
+    }
+  }
+
+  /**********\
+  |* Events *|
+  \**********/
+  access(all) event ContractCall(
+    sender: Address,
+    destinationChain: String,
+    destinationContractAddress: String,
+    payloadHash: String,
+    payload: [UInt8]
+  )
+
+  access(all) event Executed(commandId: String)
+
+  access(all) event ContractCallApproved(
+    commandId: String,
+    sourceChain: String,
+    sourceAddress: String,
+    contractAddress: String,
+    payloadHash: String,
+    sourceTxHash: String,
+    sourceEventIndex: UInt256
+  )
+
+  access(all) struct ApproveContractCallParams {
+    pub let sourceChain: String
+    pub let sourceAddress: String
+    pub let contractAddress: String
+    pub let payloadHash: String
+    pub let sourceTxHash: String
+    pub let sourceEventIndex: UInt256
+
+    init(
+      sourceChain: String,
+      sourceAddress: String,
+      contractAddress: String,
+      payloadHash: String,
+      sourceTxHash: String,
+      sourceEventIndex: UInt256,
+    ) {
+      self.sourceChain = sourceChain
+      self.sourceAddress = sourceAddress
+      self.contractAddress = contractAddress
+      self.payloadHash = payloadHash
+      self.sourceTxHash = sourceTxHash
+      self.sourceEventIndex = sourceEventIndex
+    }
+  }
+
+  access(all) let executedCommandIds: [String]
+  access(self) let approvedCommands: @{String: CGPCommand}
```

Comment:
Do we have a reference from other chains as to the scale of values this field will store? Mappings can introduce scaling considerations that can render the field (and therefore the contract) unusable if not considered.

This is especially important since the contract is permissionless and inexpensive would could present a significant spam vector.

I'm also wondering that the command id is since it appears from `execute` that this mapping is indexed on command ids. How are collisions prevented here?